src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -851,21 +851,11 @@ void FancyZones::RegisterVirtualDesktopUpdates(std::vector<GUID>& ids) noexcept
 {
     std::unique_lock writeLock(m_lock);
 
-    std::vector<GUID> deleted{};
-    m_workAreaHandler.RegisterUpdates(ids, deleted);
-
-    bool modified{ false };
-    for (const auto& id : deleted)
-    {
-        wil::unique_cotaskmem_string virtualDesktopId;
-        if (SUCCEEDED(StringFromCLSID(id, &virtualDesktopId)))
-        {
-            modified |= JSONHelpers::FancyZonesDataInstance().RemoveDevicesByVirtualDesktopId(virtualDesktopId.get());
-        }
-    }
-    if (modified)
+    m_workAreaHandler.RegisterUpdates(ids);
+    std::vector<std::wstring> active{};
+    if (VirtualDesktopUtils::GetVirtualDesktopIds(active))

```

Comment:
We should really plan to make work area unique identifier (`<device_id>_<resolution>_<desktop_d>`) as structure, not serialized string (of course, add method to serialize it). By doing so we would avoid this double calls for desktop ids, once with `GUID` and other with `std::wstring`.

Reply:
Comply agree with this, FancyZones are working with GUID (as it should be), JSONHelpers uses std::wstring and we are making conversions without actual need for it. Conversion should be made only at the very end, before storing data in file.

Reply:
@stefansjfw can you open an issue with the proposal for the work area identifier? Thanks.

Reply:
Done
https://github.com/microsoft/PowerToys/issues/3959